File: Part8/C++/FunctionConcepts/FunctionConcepts_2.cpp
// An example of function variable scope
#include <iostream>
#include <cstdio>
using namespace std;

int GlobalVariable = 15;
int GlobalToBeShadowed = 5;

void myFunction ()
{
    int myVariable = 55;
    GlobalVariable = 42;
    int globalToBeShadowed = 15;
}

int main ()
{
    printf ( "Global variable: %d\n", GlobalVariable );
    printf ( "Global shadowed: %d\n", GlobalToBeShadowed );
    myFunction ();
    printf ( "Function variable: %d\n", myVariable );
    printf ( "Global variable: %d\n", GlobalVariable );
    printf ( "Global shadowed: %d\n", GlobalToBeShadowed );
    return 0;
}

File: Part8/Rust/FunctionConcepts/FunctionConcepts_2.rs
// An example of function variable scope
static mut global_variable_glob: i32 = 15;
static mut global_to_be_shadowed_glob: i32 = 5;

fn my_function ()
{
    let my_variable = 55;
    unsafe
    {
        global_variable_glob = 42;
    }
    let global_to_be_shadowed_glob = 15;
}

fn main ()
{
    unsafe
    {
        println! ( "Global variable: {}", global_variable_glob );
        println! ( "Global shadowed: {}", global_to_be_shadowed_glob );
    }
    my_function ();
    println! ( "Function variable: {}", my_variable );
    unsafe
    {
        println! ( "Global variable: {}", global_variable_glob );
        println! ( "Global shadowed: {}", global_to_be_shadowed_glob );
    }
}
